src/Language/PureScript/Environment.hs

```diff
@@ -241,6 +241,8 @@ primTypes =
     , (primName "Boolean",  (Star, ExternData))
     , (primName "Partial",  (Star, ExternData))
     , (primName "Fail",     (FunKind Symbol Star, ExternData))
+    , (primName "TypeString", (FunKind Star Symbol, ExternData))
+    , (primName "TypeConcat", (FunKind Symbol Symbol, ExternData))
```

Comment:
That is weird indeed, total typo, should be (Funkind Symbol (Funkind Symbol Symbol)). I'll walk through the kind checker to see why it's not picking this up.